azure/services/scalesets/spec.go

```diff
@@ -132,6 +135,15 @@ func (s *ScaleSetSpec) existingParameters(ctx context.Context, existing interfac
 		return nil, nil
 	}
 
+	// if there are no model changes and no change in custom data, remove VirtualMachineProfile to avoid unnecessary VMSS model
+	// updates.
+	if !hasModelChanges && !s.ShouldPatchCustomData {
+		log.V(4).Info("removing virtual machine profile from parameters", "hasModelChanges", hasModelChanges, "shouldPatchCustomData", s.ShouldPatchCustomData)
+		vmss.Properties.VirtualMachineProfile = nil
```

Comment:
Also, how different is the line 133-136 above from the current addition ?
```
if *vmss.SKU.Capacity <= existingInfraVMSS.Capacity && !hasModelChanges && !s.ShouldPatchCustomData {
		// up to date, nothing to do
 		return nil, nil
 	}
```

Reply:
> Wont setting `vmss.Properties.VirtualMachineProfile` to `nil` cause a reconciliation nonetheless ?

a reconciliation of the AzureMachinePool? 
No, `vmss` is purely a SDK-related variable (of type `armcompute.VirtualMachineScaleSet`) and is used by `async` ResourceGetters to construct the parameters for making an Azure API call. 

Let me know if I misunderstood your question. 

> Also, how different is the line 133-136 above from the current addition ?

Good question. 

line 133-136 avoids making an API call (because of returning nil, nil) altogether when there's no increase in replicas (and no model changes needed). 

the new addition ensures there's no VMSS model change when only replica count is increased. 

Does that make sense?

Reply:
Thank you for the response, makes sense to me.